src/bf_dpu_update.py

```diff
@@ -1485,6 +1485,12 @@ def update_bundle(self):
         cur_vers = self._get_all_versions_internal()
         old_bmc_ver = cur_vers['BMC']
 
+        # Ensure golden image config DIR exist before config update procedure
+        if self.reset_bios and not self.ensure_golden_image_config_dir_on_bmc():
+            raise Err_Exception(Err_Num.BMC_GOLDEN_IMAGE_CONFIG_DIR_ERR,
+                'Please make sure /tmp/golden-image-config exist on BMC'
+            )
+
```

Comment:
Stephen, why do you need this "if" here? If you in the bundle update flow (BFB) then BFB takes care of the /tmp/golden-image-config. Did you see any issue here? I will approve this change but for the bundle/fwbundle BFB it is not needed.